lib/provider.js

```diff
@@ -519,8 +525,14 @@ export default function() {
             let currentStateTransition = angular.extend({}, $state.current);
             let toStateTransition = angular.extend({}, $state.get(stateName));
             $log.debug('nativepagetransitions goBack', backCount, stateName, currentStateTransition, toStateTransition);
-            $ionicHistory.goBack(backCount);
             transition('back', currentStateTransition, toStateTransition);
+            if (ionic.Platform.isIOS()) {
```

Comment:
might be useful to have the timeout also for Android. I had transition issues from time to time on Android and it seems like timeout makes it better.